### DISCUSSION TITLE
Usage of ProjectionAux for the whole domain

### DISCUSSION THREAD

**Discussion by rnpratoori** [idaholab/moose#28313] [Q&A Modules: General] [answered]
Hello,

I am currently using `ProjectionAux` to calculate the average of a variable for each element. Is there a way I can use it for all the elements - I need the average of the value for all the elements.

Just FYI - I know the average value (it is one of the constant global values in the input file). I need it to be used in a `AuxKernel` I wrote. This value does not change throughout the simulation. So, any other alternative way of handling it is also appreciated.

Thanks in advance,
Raghunandan.

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can set an auxvariable using a ParsedAux if you already know the value.
Computing a single value for the entire domain is more the role of a Postprocessor than an auxkernel. You should use something like this
https://mooseframework.inl.gov/source/postprocessors/ElementAverageValue.html
then use a ParsedAux to spread the value to a field, only if it's really needed
